File: speeding_fine_tshirt/Graph/ScriptTemplate/Pack/CGPackNode.js
/**
 * @file CGPackNode.js
 * @author Weifeng Huang (huangweifeng.2067@bytedance.com)
 * @date 2021-08-26
 * @brief
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */
const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGPackNode extends BaseNode {
  constructor() {
    super();
    this.valueType = null;
  }

  getOutput(index) {
    if (this.valueType === 'Vector2f') {
      return new Amaz.Vector2f(this.inputs[0](), this.inputs[1]());
    } else if (this.valueType === 'Vector3f') {
      return new Amaz.Vector3f(this.inputs[0](), this.inputs[1](), this.inputs[2]());
    } else if (this.valueType === 'Vector4f') {
      return new Amaz.Vector4f(this.inputs[0](), this.inputs[1](), this.inputs[2](), this.inputs[3]());
    } else if (this.valueType === 'Quaternionf') {
      return new Amaz.Quaternionf(this.inputs[0](), this.inputs[1](), this.inputs[2](), this.inputs[3]());
    } else if (this.valueType === 'Rect') {
      return new Amaz.Rect(this.inputs[0](), this.inputs[1](), this.inputs[2](), this.inputs[3]());
    } else if (this.valueType === 'Color') {
      return new Amaz.Color(this.inputs[0](), this.inputs[1](), this.inputs[2](), this.inputs[3]());
    }
  }
}

exports.CGPackNode = CGPackNode;

File: speeding_fine_tshirt/Graph/ScriptTemplate/Pack/CGPackNode.lua
local CGPackNode = CGPackNode or {}
CGPackNode.__index = CGPackNode

function CGPackNode.new()
    local self = setmetatable({}, CGPackNode)
    self.inputs = {}
    self.valueType = nil
    return self
end

function CGPackNode:setInput(index, func)
    self.inputs[index] = func
end

function CGPackNode:getOutput(index)
    if self.valueType == "Vector2f" then
        return Amaz.Vector2f(self.inputs[0](), self.inputs[1]())
    elseif self.valueType == "Vector3f" then
        return Amaz.Vector3f(self.inputs[0](), self.inputs[1](), self.inputs[2]())
    elseif self.valueType == "Vector4f" then
        return Amaz.Vector4f(self.inputs[0](), self.inputs[1](), self.inputs[2](), self.inputs[3]())
    elseif self.valueType == "Quaternionf" then
        return Amaz.Quaternionf(self.inputs[0](), self.inputs[1](), self.inputs[2](), self.inputs[3]())
    elseif self.valueType == "Rect" then
        return Amaz.Rect(self.inputs[0](), self.inputs[1](), self.inputs[2](), self.inputs[3]())
    elseif self.valueType == 'Color' then
        return Amaz.Color(self.inputs[0](), self.inputs[1](), self.inputs[2](), self.inputs[3]())
    end
end

return CGPackNode
